src/arrayMethodJoin.js

```diff
@@ -5,7 +5,21 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let resultingString = '';
+    const splitter = separator === undefined ? ',' : separator;
+
+    for (let i = 0; i < this.length; i += 1) {
```

Comment:
```suggestion
    for (let i = 0; i < this.length; i ++) {
```
use increment in this case